### ISSUE TITLE
Use QScintilla as the editor

### ISSUE THREAD

**Issue by ouuan** [coder3101/cp-editor#108]
**Is your feature request related to a problem? Please describe.**

QCodeEditor has many problems, see the [issues](https://github.com/Megaxela/QCodeEditor/issues).

**Describe the solution you'd like**

Use QScintilla as the editor.

**Describe alternatives you've considered**

Improve QCodeEditor.

**Additional context**

It's been discussed on Telegram, open here just for tracking.

**Comment by ouuan** [comment]
In fact, I'd like to write a Chinese blog to introduce CP Editor to others as soon as the new editor is done. I think QCodeEditor has too many problems that I can't persuade people to use CP Editor.

**Comment by coder3101** [comment]
Yeah sure. I will surely work on implementing this ASAP.

**Comment by coder3101** [comment]
Work in Progress : On branch scintilla. 

There are some features that may not work with scintilla like themes, so for now scintilla will only have Default white theme

**Comment by ouuan** [comment]
We can add themes manually, though that's a big work. I think it's necessary to support at least one dark theme first.

**Comment by coder3101** [comment]
Yeah, first let me make it fully functional then we will work on UI aspect of it

**Comment by coder3101** [comment]
QScintilla will be an addition and QCodeEditor will not go anywhere. User will have choice to use one of the two editors, however change of editor will require the restart to take effect.

**Comment by ouuan** [comment]
![image](https://user-images.githubusercontent.com/30581822/73366641-3e676c80-42e9-11ea-965b-fee361a4e989.png)

![image](https://user-images.githubusercontent.com/30581822/73366663-47f0d480-42e9-11ea-8d4b-1f014a7bcdc6.png)

(from Telegram private messages)

**Comment by coder3101** [comment]
This needs to become optional, I will still work on this and add as optional feature.

**Comment by coder3101** [comment]
**We are continuing to use this editor but since language server will offer auto-complete and other stuff, it is better that our editor gets these features now**
